sql-plugin/src/main/scala/com/nvidia/spark/rapids/CostBasedOptimizer.scala

```diff
@@ -351,6 +359,13 @@ object MemoryCostHelper {
   def calculateCost(dataSize: Long, memorySpeed: Double): Double = {
     (dataSize / GIGABYTE) / memorySpeed
   }
+
+  def isWindowExpr[INPUT <: Expression](expr: BaseExprMeta[INPUT]) = {
```

Comment:
Can we change the name of this? A `WindowExpression` is something different that does have a type. It has children that might now, but the name to be feels broken.

Reply:
Would `isUnevaluableWindowExpr` be appropriate?

Reply:
Maybe this method can be a catch all for all expressions whose cost cannot be evaluated, and the name should not refer to a specific flavor of expressions?